ai_ref_kits/multimodal_ai_visual_generator/main.py

```diff
@@ -1,17 +1,188 @@
+llm_config.apply_chat_template = False
+
+# ---------- Request/Response Schemas ----------
+class PromptRequest(BaseModel):
+    prompt: str
+
+class StoryRequest(BaseModel):
+    prompt: str
+
+# ---------- LLM Endpoint (Story Splitter) ---------
+@app.post("/generate_story_prompts")
+def generate_story_prompts(request: StoryRequest, req: Request):
+    if not llm_pipe:
+        return JSONResponse(status_code=503, content={"error": "LLM model not available. Please export it before using this endpoint."})
+    config_type = req.query_params.get("config", "illustration")
+    config_file = PROJECT_ROOT / "config" / f"{config_type}.yaml"
+    if not config_file.exists():
+        raise RuntimeError(f"Config file not found: {config_file}")
+    with open(config_file, "r") as f:
+        config = yaml.safe_load(f)
+
+    user_prompt = request.prompt
+
+    instruction = config["instruction_template"].replace("{user_prompt}", user_prompt)
+
+    output = []
+    def streamer(subword):
+        sys.stdout.write(subword)
+        sys.stdout.flush()
+        output.append(subword)
+        return False
+
+    _ = llm_pipe.generate(instruction, llm_config, streamer)
+    full_output = "".join(output)
+
+    # --- Parse scenes ---
+    scenes = []
+    current_scene = ""
+    for line in full_output.split("\n"):
+        clean = line.strip()
+        if clean.lower().startswith(config["scene_prefix"].lower()):
+            if current_scene:
+                scenes.append(current_scene.strip())
+            parts = clean.split(":", 1)
+            if len(parts) == 2:
+                current_scene = parts[1].strip()
+        elif clean:
+            current_scene += " " + clean
+
+    if current_scene:
+        scenes.append(current_scene.strip())
+
+    # --- Clean & format output ---
+    suffixes = config["scene_suffixes"]
+    max_words = config["max_words_per_scene"]
+    final_scenes = []
+    for i, scene in enumerate(scenes[:4]):
+        words = scene.split()
+        trimmed = " ".join(words[:max_words])
+        suffix = suffixes[i] if i < len(suffixes) else suffixes[-1]
+        if suffix.lower() not in trimmed.lower():
+            if not trimmed.endswith((".", "!", "?")):
+                trimmed += "."
+            trimmed += " " + suffix
+        final_scenes.append(trimmed.strip())
+
+    while len(final_scenes) < 4:
+        fallback_suffix = suffixes[len(final_scenes)] if len(suffixes) > len(final_scenes) else suffixes[-1]
+        final_scenes.append(config["fallback_scene"] + ". " + fallback_suffix)
+   
+    return {"scenes": final_scenes}
+
+# ---------- Image Model Endpoint (Image Generator) ----------
+@app.post("/generate_images")
+def generate_image(request: PromptRequest):
+    if not image_pipe:
+        return JSONResponse(status_code=503, content={"error": "Image model not available. Please export it before using this endpoint."})
+
+    prompt = request.prompt
+    height = 512
+    width = 512
+    seed = 191524753

```

Comment:
Shouldn't be random?

Reply:
Replaced fixed seed with a 32-bit random seed and logged it.